main.cc

```diff
@@ -186,6 +186,12 @@ struct convert<::object_storage_endpoint_param> {
             ep.config.https.emplace();
             ep.config.https->trust_file = node["https_truststore"].as<std::string>();
         }
+        if (node["aws_region"]) {
```

Comment:
question about the commit message (since I can't write a question there): you said "The content can be unsigned when sent over HTTPS, this is what this patch does.", but:
1. I don't see where the patch does this (it seems it always adds the signature?)
2. I still don't understand how our s3 bucket can be secure if we don't force s3 to check for signed requests, even on https. I guess you are trusting specific source addresses, but I don't think this is a very secure setup (even if better than nothing).

Reply:
I wasn't clear enough, sorry, let me clarify.

The patch does signs the request, without signing AWS doesn't let multipart upload to work. However, when signing the request there are many options and requirements how to do that and one them concerns the message body. That's called "payload singing" in the docs. Signing the payload is sometimes optional, even for multipart upload, and this is what I do here -- I sign the request _without_ the body, only three headers and that's it.

Reply:
(updated the PR description text)

Reply:
I understand now.

One last thought on this topic: If you don't do "payload signing", what prevents an attacker from reusing the same signed headers with a different payload? A sort of a "replay attack" but with a different payload? I'm guessing that the idea is that if we use HTTPS, attackers cannot read any of the signed headers, so they cannot replay anything. 

(In this case, Amazon could make things even more efficient, and require that just the first request on an HTTPS connection be signed, there is no point in signing the next requests without signing the payloads, it adds no security. But maybe they still insist on these signatures, just in case each request has different credentials...).

Reply:
> One last thought on this topic: If you don't do "payload signing", what prevents an attacker from reusing the same signed headers with a different payload? A sort of a "replay attack" but with a different payload? I'm guessing that the idea is that if we use HTTPS, attackers cannot read any of the signed headers, so they cannot replay anything.

Yes, it's HTTPS that's relied on in this case. When using plain HTTP AWS requires that the payload is also signed, not only some headers.